modules/azure_devops_agent_container_app_jobs/keyvault.tf

```diff
@@ -0,0 +1,59 @@
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "agent_vault" {
+  name                       = "${var.resource_prefix}${random_string.name.result}agent"
```

Comment:
```suggestion
  name                       = "${var.resource_prefix}-${random_string.name.result}-agent"
```

Reply:
keyvaults have a name length restriction of 24 chars so I chose to not include dashes. The random part takes 6 chars leaving a max prefix length of 18 adding dashes reduces it to 16.